middleman-core/lib/middleman-core/server_listener.rb

```diff
@@ -0,0 +1,72 @@
+require 'socket'
+require 'resolv'
+
+module Middleman
+  class ServerListener
+    private
+
+    attr_reader :https
+
+    public
+
+    attr_reader :bind_address, :server_name, :port, :all_available_network_interfaces
+
+    def initialize(opts)
+      #@bind_address = opts[:bind_address]
+      #@server_name  = opts[:server_name]
+
+      # fail ArgumentError, 'You either need to define "bind_address" or "server_name".' if !bind_address && !server_name
+      # fail ArgumentError, 'You can only define "bind_address" or "server_name". Not  both' if bind_address && server_name
+      # fail ArgumentError, 'Server Name resolve to a locally available network interface.' unless all_available_network_interfaces.include?(resolve_ip(@server_name))
+
+      # '0.0.0.0'
+      # '127.0.0.1'
+
+      #@bind_address = resolve_ip(@server_name)
+
+      #@port  = opts.fetch(:port)
+      #@https = opts.fetch(:https)
+
+
+      # @all_available_network_interfaces = find_network_interfaces
+    end
+
+    def to_listeners
+    #   return [uri(bind_address)] if bind_address
+    #   return [uri(server_name)] if server_name
+
+    #   all_listeners.map { |l| uri(l) }
+    end
+
+    private
+
+    def uri(sn)
+      scheme = https? ? 'https' : 'http'
+      URI("#{scheme}://#{sn}:#{port}/")
+    end
+
+    def find_network_interfaces
+      Socket.ip_address_list.find(-> { ['127.0.0.1'] }) { |ai| ai.ipv4? && !ai.ipv4_loopback? }.map { |ai| ai.ip_address }
```

Comment:
I should have mentioned, that the code is very very broken and only the tests should be reviewed. Sorry for being that unprecise.